packages/@jsii/runtime/lib/sync-stdio.ts

```diff
@@ -65,3 +65,58 @@ export class SyncStdio {
     }
   }
 }
+
+/**
+ * A patched up implementation of `fs.readSync` that works around the quirks associated with
+ * synchronous I/O involving `STDIN`, `STDOUT` and `STDERR` on NodeJS, where this is all expected
+ * to be asynchronous and hence has some "interesting" behavior in certain particular cases.
+ *
+ * @param fd        the file descriptor to read from (typically 0 / STDIN)
+ * @param buffer    the buffer in which to place the read data
+ * @param offset    the offset in the buffer at which to place the read data
+ * @param length    the number of bytes to be read
+ * @param position  where to begin reading from the file (defaults to the current read location)
+ *
+ * @returns the amount of bytes read, or `0` if EOF has been reached.
+ */
+function readSync(
+  fd: number,
+  buffer: Buffer,
+  offset: number,
+  length: number,
+  position: number | null = null,
+): number {
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
```

Comment:
I used a `while` loop instead of recursively re-entering, as this avoids stacking up in case `EAGAIN` happens *a lot*, which I imagine might be somewhat frequent on resource-constrained systems (where we'd likely also have less memory available for wasting on stack frames).

Reply:
This might be a nice comment to have in source, rather than just on the PR.